rules/table-of-contents.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+const visit = require('unist-util-visit');
+const chalk = require('chalk');
+
+module.exports = (ast, file, language, done) => {
+	visit(ast, 'heading', (node, index) => {
+		const title = node.children[0].value;
+
+		if (index === 1 && title !== 'Contents') {
```

Comment:
What's the advantage of moving from `unist-util-visit` to a `filter`? If I understood, you are saying that `<h1>`s aren't parsed by `unist`? 🤔

Thanks for noticing that I'm not checking if the `title` is `undefined` 😅 Will fix it 💪 

Now, a question (for @sindresorhus and you as well): should we support things like `<h2> Contents` and `## **Contents**`?